pkg/operator/reports_test.go

```diff
@@ -92,3 +101,190 @@ func TestGetNextReportPeriod(t *testing.T) {
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 2, 0)},
+			}, schedule, false),
+			expectRunning: true,
+		},
+		{
+			name: "spec.Reason set to 'Scheduled' returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "spec.Reason set to 'InvalidReport' returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "spec.Status set to v1.ConditionTrue returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+	}
+
+	for _, testCase := range testTable {
+		var mockLogger = logrus.New()
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			runningCond := isReportFinished(mockLogger, testCase.report)
+			assert.Equalf(t, runningCond, testCase.expectRunning, "expected the report '%s' would return '%t', but got '%t'", testCase.name, testCase.expectRunning, runningCond)
+		})
+	}
+}
+
+func TestValidateReport(t *testing.T) {
+	const (
+		testNamespace           = "default"
+		testReportName          = "test-report"
+		testQueryName           = "test-query"
+		testGenerationQueryName = "test-GQN"
+	)
+
+	ds1 := testhelpers.NewReportDataSource("datasource1", testNamespace)
+
+	testQuery := &metering.ReportGenerationQuery{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-query",
+			Namespace: testNamespace,
+		},
+		Spec: metering.ReportGenerationQuerySpec{
+			DataSources: []string{
+				"datasource1",
+			},
+		},
+	}
+
+	dataSourceGetter := testhelpers.NewReportDataSourceStore([]*metering.ReportDataSource{ds1})
+	queryGetter := testhelpers.NewReportGenerationQueryStore([]*metering.ReportGenerationQuery{testQuery})
+	reportGetter := testhelpers.NewReportStore(nil)
+
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	testTable := []struct {
+		name         string
+		report       *metering.Report
+		queryName    string
+		expectErr    bool
+		expectErrMsg string
+	}{
+		{
```

Comment:
Need a test case for the happy path when ValidateReport doesn't return an error.

Reply:
Not seeing this fixed @timflannagan1

Reply:
Also a test case where the error is a dependency error, eg, when a generationQuery depends on a non-existent datasource.